src/Polytopes/Polyhedron.jl

```diff
@@ -0,0 +1,324 @@
+@doc Markdown.doc"""
+    Polyhedron(A, b)
+
+The (metric) polyhedron defined by
+
+$$P(A,b) = \{ x |  Ax ≤ b \}.$$
+
+see Def. 3.35 and Section 4.1.
+""" struct Polyhedron #a real polymake polyhedron
+    pm_polytope::Polymake.BigObjectAllocated
+    boundedness::Symbol # Values: :unknown, :bounded, :unbounded
+end
+function Polyhedron(pm_polytope::Polymake.BigObjectAllocated)
+    Polyhedron(pm_polytope, :unknown)
+end
+function Polyhedron(A::Union{Oscar.MatElem,AbstractMatrix}, b)
+    Polyhedron(Polymake.polytope.Polytope{Polymake.Rational}(
+        INEQUALITIES = matrix_for_polymake(remove_zero_rows([b -A])),
+    ))
+end
+
+"""
+    pm_polytope(P::Polyhedron)
+
+Get the underlying polymake `Polytope`.
+"""
+pm_polytope(P::Polyhedron) = P.pm_polytope
+
+property_is_computed(P::Polyhedron, S::Symbol) = property_is_computed(pm_polytope(P), S)
+==(P0::Polyhedron, P1::Polyhedron) = Polymake.polytope.equal_polyhedra(pm_polytope(P0), pm_polytope(P1))
+
+###############################################################################
+###############################################################################
+### Display
+###############################################################################
+###############################################################################
+function Base.show(io::IO, P::Polyhedron)
+    print(io, "A Polyhedron of dimension $(dim(P))")
+end
+
+
+
+###############################################################################
+###############################################################################
+### Iterators
+###############################################################################
+###############################################################################
+
+
+struct VertexPointIterator
+    p::Polyhedron
+end
+
+function Base.iterate(iter::VertexPointIterator, index = 1)
+    vertices = pm_polytope(iter.p).VERTICES
+    while true
+        if size(vertices, 1) < index
+            return nothing
+        end
+
+        if iszero(vertices[index, 1])
+            index += 1
+        else
+            return (vertices[index, 2:end], index + 1)
+        end
+    end
+end
+Base.eltype(::Type{VertexPointIterator}) = Polymake.VectorAllocated{Polymake.Rational}
+Base.length(iter::VertexPointIterator) = n_vertices(iter.p)
+
+"""
+   vertices(H::Polyhedron, as = :points)
+
+Returns the vertices of a polyhedron.
+"""
+function vertices(P::Polyhedron; as = :points)
+    if as == :points
+        VertexPointIterator(P)
+    elseif as == :point_matrix
+        return(decompose_vdata(pm_polytope(P).VERTICES).vertices)
```

Comment:
It's a bit odd that this code once uses an implicit return, then an explicit one.